ddtrace/span.py

```diff
@@ -336,9 +336,12 @@ def _remove_tag(self, key):
             del self.meta[key]
 
     def get_tag(self, key):
-        # type: (_TagNameType) -> Optional[Text]
+        # type: (_TagNameType) -> Any
```

Comment:
This should be a union of either what tag or metric could be and not `Any`. That is too permissive.

Reply:
I set the return type to `Any` because set_tag accepts `Any` value. After taking a second look at `set_tag` I can see it attempts to stringify the value. If we move forward with this change I'll add this fix